src/acceleration/BaseQNAcceleration.cpp

```diff
@@ -149,24 +166,15 @@ void BaseQNAcceleration::initialize(
     }
 
     // test that the computed number of unknown per proc equals the number of entries actually present on that proc
-    size_t unknowns = _dimOffsets[utils::IntraComm::getRank() + 1] - _dimOffsets[utils::IntraComm::getRank()];
-    PRECICE_ASSERT(entries == unknowns, entries, unknowns);
+    // size_t unknowns = _dimOffsets[utils::IntraComm::getRank() + 1] - _dimOffsets[utils::IntraComm::getRank()];
+    // PRECICE_ASSERT(entries == unknowns, entries, unknowns); //?
```

Comment:
This is a tricky point. Should `_dimOffsets` include secondary data or not? Probably makes a difference for IMVJ. Let's try to also add a test with secondary data that runs in parallel.

Reply:
Since it's used to initialize the size of `Wtil`, I would say that the `_dimOffsets` should include the secondary data. But I'm a little confused with this function now: https://github.com/precice/precice/blob/8458db5189c7f4c9c41551cac55eb957ce46af80/src/acceleration/BaseQNAcceleration.cpp#L626-L632
In the extended tests, the parallel tests with IMVJ include also secondary data. They could work with this implementation here(secondary data included in the `_dimOffsets`). This assert should be modified a bit then.